src/Inputs/ComboBox.tsx

```diff
@@ -0,0 +1,295 @@
+import React, {
+    useState,
+    useEffect,
+    useMemo,
+    useCallback,
+    Children,
+    isValidElement,
+} from 'react';
+import styled, { withTheme, DefaultTheme } from 'styled-components';
+import {
+    flex,
+    scroll,
+    position,
+    transition,
+    clickable,
+    styledCondition,
+    darken,
+} from '@Utils/Mixins';
+import { useTransition } from '@Utils/Hooks';
+import { LabelLayout, LabelLayoutProps,InputFragment } from '@Layouts';
+ 
+const NUM_VISIBLE_SELECTIONS=3;
+
+const SPEED = 'normal';
+
+let myWindow = window as any;
+
+export interface SelectProps extends LabelLayoutProps {
+    disabled?: boolean;
+    placeholder?: string;
+    value?: string | number;
+    theme: DefaultTheme;
+    onChange?: Function;
+    limit?: number;
+    numVisible:number;
+}
+
+const _Select: React.FC<SelectProps> = ({
+    disabled,
+    value,
+    children,
+    limit = 4,
+    placeholder = '',
+    onChange = (): void => {},
+    theme,
+    name,
+    numVisible=NUM_VISIBLE_SELECTIONS,
+    ...props
+}): React.ReactElement => {
+    const [expanded, setExpanded] = useState(false);
+    const options = Children.toArray(children);

```

Comment:
This will run every render, should this be in a useEffect?

Reply:
useMemo